### DISCUSSION TITLE
Solving for a periodic variable

### DISCUSSION THREAD

**Discussion by gj19866** [idaholab/moose#26707] [Q&A Getting Started]
Hi there,

I am having to reconsider some of the methods I am using in my model. 
Instead of considering a complex variable $\psi$ as $\psi = \psi_R + i \psi_I$, I would like to consider it as the form $\psi = |\psi|e^{i\chi}$.
Here I would have to solve for the variables $|\psi|$, and $\chi$. 

Before I dive into this fully, do you think that moose would be able to effectively solve for a variable $\chi$ that exists in a range $(-\pi, \pi]$. 
I am worried about the discontinuity, and that $\chi = \chi +2\pi$ in the solve. 

What are your thoughts about this?

Many thanks,
Gillian

**Comment by GiudGiud** [comment]
Hello

I would be worried about two things:
- the singularity near psi = 0 because of the absolute value
- the need for something to pin the chi between 0 and 2 pi, the value could drift without changing the solution due to periodicity

I would go ahead and implement it and find remedies for both if needed.
I think the first one we can use the Bounds system
For the second one we can probably constrain the average of the variable chi over the domain

Guillaume

**Reply by gj19866** [reply]
Amazing, Ill give it a shot and see what happens thanks!

**Reply by GiudGiud** [reply]
@cticenhour @aaelmeli if you have experience with working with complex variable in this (module, angle) form please let us know!

**Reply by cticenhour** [reply]
I do not have experience working with this form of complex numbers in MOOSE. Your suggestions seem reasonable to start, @GiudGiud.
